### ISSUE TITLE
How to bundle pure JS and CSS separately without HTML

### ISSUE THREAD

**Issue by sque89** [parcel-bundler/parcel#4061]
# ❔ Question

I want to use parcel to bundle JS and CSS of a wordpress theme. As that isn't a app with a index.html which could be used as an entry point, i am struggling how to proper bundle these assets.

So i added two targets, one for JS and one for CSS. The build is running fine, the CSS bundle is working fine. Basically the JS bundle also looks fine but if i am running it in the browser, i am getting this error:

**Uncaught ReferenceError: require is not defined**

## 🔦 Context

We want to introduce parcel as out build tool to keep it simple.

## 💻 Code Sample

**package.json**

`"js": "../public/build.js",
  "css": "../public/build.css",
  "targets": {
    "js": {
      "publicUrl": "/app/themes/wp-theme-us/public/",
      "context": "browser"
    },
    "css": {
      "publicUrl": "/app/themes/wp-theme-us/public/",
      "context": "browser"
    }
  },
  "scripts": {
    "watch-js": "parcel watch js/index.js --target js",
    "watch-css": "parcel watch css/index.css --target css",
    "build-js": "parcel build js/index.js --target js",
    "build-css": "parcel build css/index.css --target css",
    "build": "rm -rf ../public/ && npm install && npm run build-js && npm run build-css && cp -r ../assets/. ../public",
    "watch": "npm install && concurrently \"npm run watch-js\" \"npm run watch-css\"",
    "lint-css": "stylelint \"css/**/*.css\""
  }`

babel.config.js

`module.exports = {
  "plugins": [
    "@babel/plugin-transform-object-assign"
  ],
  "presets": [
    [
      "@babel/preset-env",
      {
        "useBuiltIns": "usage",
        "include": [
          "es6.object.assign"
        ]
      }
    ]
  ],
  // babelify is used with `--global` option to also transpile language files of
  // numbro. here, we ignore everything else then numbro to prevent collateral
  // damage, e.g. in core-js.
  "ignore":  [
    /node_modules\/(?!numbro\/)/
  ]
}
`
## 🌍 Your Environment

| Software         | Version(s) |
| ---------------- | ---------- |
| Parcel           | 2.0.0-alpha.3.2
| Node             |11.15.0
| npm/Yarn         |6.7.0
| Operating System |Linux

**Comment by mischnic** [comment]
This should work, please provide a reproduction for the error (a code sample for index.js).

**Comment by sque89** [comment]
Hi @mischnic, thanks so much for the fast reply. I tracked the problem down to a 3rd-party-library called numbro (https://github.com/BenjaminVanRyseghem/numbro).

I added a link to the issue description an example project on github where i reproduce the problem with a minimal version. If numbro gets required, the javascript is not running anymore.

**Comment by mischnic** [comment]
*Might* be a duplicate of https://github.com/parcel-bundler/parcel/issues/4003#issuecomment-575318781

**Comment by sque89** [comment]
The line in question looks different to the one in #4003 but i just had a look at the minified version. Is there a way to build without minification? Then i could further inspect if it is the same.

So #4003 is an actual bug which needs to be fixed in parcel? Is there any workaround by config or so?

**Comment by mischnic** [comment]
`--no-minify`

That specific one inserted `require("process")`, there is a PR to fix it.

**Comment by sque89** [comment]
I dont think it is a duplicate. The line which is causing the problem is included in the numbro bundle.

`for (var l = "function" == "function" && require, e = 0; e < c.length; e++) s(c[e]);`

If i search for this line in the original numbro.min.js (which works if included directly in the HTML) the i find it but slightly different:

`for (var l = "function" == typeof require && require, e = 0; e < c.length; e++) s(c[e]);`

Could this be a problem with bundling an existing bundling again or so?

**Comment by mischnic** [comment]
Yes, this is a different problem.

> Could this be a problem with bundling an existing bundle again or so?

This is the reason, but this should work.
